ping/ping.c

```diff
@@ -82,6 +82,36 @@ ping_func_set_st ping4_func_set = {
 #define	NROUTES		9		/* number of record route slots */
 #define TOS_MAX		255		/* 8-bit TOS field */
 
+#define CASE_TYPE(x) case x: return #x;
+
+static char *str_family(int family)
+{
+	switch (family) {
+		CASE_TYPE(AF_UNSPEC)
+		CASE_TYPE(AF_INET)
+		CASE_TYPE(AF_INET6)
+	default:
+		error(2, 0, _("unknown protocol family: %d"), family);
+	}
+
+	return NULL;
+}
+
+static char *str_socktype(int socktype)
+{
+	if (!socktype)
+		return "0";
+
+	switch (socktype) {
+		CASE_TYPE(SOCK_DGRAM)
+		CASE_TYPE(SOCK_RAW)
+	default:
+		error(2, 0, _("unknown sock type: %d"), socktype);
+	}
+
+	return NULL;
```

Comment:
We still return NULL here. How about just ""?

Reply:
Thanks, makes sense, will update that before merge.